Client/game_sa/CModelInfoSA.cpp

```diff
@@ -1407,6 +1407,24 @@ void CModelInfoSA::MakeObjectModel(ushort usBaseID)
     CopyStreamingInfoFromModel(usBaseID);
 }
 
+void CModelInfoSA::MakeTimedObjectModel(ushort usBaseID)
+{
+    CTimeModelInfoSAInterface* m_pInterface = new CTimeModelInfoSAInterface();
+
```

Comment:
Does the memory get deallocated?

Reply:
Apparently, all current `CModelInfoSA::Make*` functions cause a memory leak. The fix is simple, we just need to call delete in `CModelInfoSA::DeallocateModel`.
```
void CModelInfoSA::DeallocateModel(void)
{
    Remove();
    delete ppModelInfo[m_dwModelID]; // <---------- This line fixes the memory leak problem.
    ppModelInfo[m_dwModelID] = nullptr;
    pGame->GetStreaming()->GetStreamingInfoFromModelId(m_dwModelID)->Reset();
}
```

Reply:
But we need select right class type here. I don't what function `Remove` doing.

Reply:
I will add `switch case` here

Reply:
> But we need select right class type here. I don't what function Remove doing.

CModelInfoSA::Remove does not delete the model info. It only deletes the copy of allocated clump. Internally, it calls CStreaming::RemoveModel. You can see the reversed code [here](https://github.com/saml1er/gta-reversed/blob/master/source/game_sa/CStreaming.cpp#L2275). So in short, the fix is simple. Since we are allocating a custom interface, then we must delete it in CModelInfoSA::DeallocateModel.

Reply:
> I will add `switch case` here

It's really not needed. It's just a pointer.

Reply:
but it's pointer to `CBaseModelInfo`. Will be `delete` with children classes?

Reply:
> 
> 
> but it's pointer to `CBaseModelInfo`. Will be `delete` works with children classes?

CBaseModelInfoSAInterface is the base class, and it has a virtual destructor, so yes, it will work.

Reply:
Done